src/app/calculator/motors/full-load-amps/full-load-amps.service.ts

```diff
@@ -0,0 +1,280 @@
+    } else {
+      return [];
+    }
+  }
+
+  getObjFromForm(form: FormGroup): FanMotor {
+    let obj: FanMotor = {
+      lineFrequency: form.controls.lineFrequency.value,
+      motorRatedPower: form.controls.motorRatedPower.value,
+      motorRpm: form.controls.motorRpm.value,
+      efficiencyClass: form.controls.efficiencyClass.value,
+      specifiedEfficiency: form.controls.specifiedEfficiency.value,
+      motorRatedVoltage: form.controls.motorRatedVoltage.value,
+      fullLoadAmps: undefined,
+    };
+    return obj;
+  }
+
+  isFanMotorValid(obj: FanMotor): boolean {
+    let form: FormGroup = this.getFormFromObj(obj);
+    return form.valid;
+  }
+
+  estimateFullLoadAmps(settings: Settings) {
+    let flaInput = this.fullLoadAmpsInputs.getValue();
+    let tmpEfficiency: number;
+
+    if (flaInput.efficiencyClass !== 3) {
+      tmpEfficiency = flaInput.efficiencyClass;
+    } else {
+      tmpEfficiency = flaInput.specifiedEfficiency;
+    }
+
+    let fullLoadAmpsResult = this.psatService.estFLA(
+      flaInput.motorRatedPower,
+      flaInput.motorRpm,
+      flaInput.lineFrequency,
+      flaInput.efficiencyClass,
+      tmpEfficiency,
+      flaInput.motorRatedVoltage,
+      settings
+    );
+
+    this.fullLoadAmpsResult.next(fullLoadAmpsResult);
+  }
+
+  changeEfficiencyClass(form: FormGroup) {
+    let tmpEfficiencyValidators: Array<ValidatorFn> = this.getEfficiencyValidators(form.controls.efficiencyClass.value);
+    form.controls.efficiencyClass.setValidators(tmpEfficiencyValidators);
+    form.controls.efficiencyClass.reset(form.controls.efficiencyClass.value);
+    form.controls.efficiencyClass.markAsDirty();
+    return form;
+  }
+
+
+  roundVal(val: number, digits: number) {
+    return Number(val.toFixed(digits))
+  }
+
+  generateExampleData(settings: Settings) {
+    let defaultData: FanMotor = {
+      lineFrequency: 60,
+      motorRatedPower: 600,
+      motorRpm: 1180,
+      efficiencyClass: 1,
+      specifiedEfficiency: 100,
+      motorRatedVoltage: 470,
+      fullLoadAmps: 683.25
+    };
+
+    
+    this.fullLoadAmpsInputs.next(defaultData);
+  }
+
+  initDefualtEmptyInputs(settings: Settings) {
+    let emptyInput: FanMotor = {
+      lineFrequency: 50,
+      motorRatedPower: 0,
+      motorRpm: 0,
+      efficiencyClass: 0,
+      specifiedEfficiency: 0,
+      motorRatedVoltage: 0,
+      fullLoadAmps: 0
+    };
+    this.fullLoadAmpsInputs.next(emptyInput);
+  }
+
+  initDefualtEmptyOutputs() {
+    this.fullLoadAmpsResult.next(undefined);
+  }
+
+
+  checkMotorWarnings(motor: FanMotor, settings: Settings): FLAMotorWarnings {
+    let efficiencyError: string = null;
+    if (motor.efficiencyClass === 3) {
+      efficiencyError = this.checkEfficiency(motor);
+    }
+    let warnings: FLAMotorWarnings ={
+      rpmError: this.checkMotorRpm(motor),
+      voltageError: this.checkMotorVoltage(motor),
+      flaError: this.checkFLA(motor, settings),
```

Comment:
Since you're defining FLAMotorWarnings interface below instead of using the warnings interface from assessment, you can remove flaError - it's not an input in this calc.

Reply:
The checkFLA method can also be removed